### ISSUE TITLE
Cannot use additional assertions

### ISSUE THREAD

**Issue by alexanderabramov** [tginsberg/junit5-system-exit#1]
I'm testing a CLI, and would like to write a test that calls `main` (which eventually calls `System.exit()`), and verifies output produced to (stubbed) `System.out` and calls made to mocked dependencies.

I've tried this approach (Kotlin):
```
assertThrows<com.ginsberg.junit.exit.SystemExitPreventedException> {
    main(arrayOf())
}

assertTrue(something else)

throw SystemExitPreventedException()
```

Unfortunately this does not compile since `SystemExitPreventedException` is package-private.

Would it be possible to make the exception public, or provide for a different mechanism for assertions after `exit()` is prevented?

**Comment by tginsberg** [comment]
Yikes! Didn't realize this issue was here, I'm sorry for the VERY long delay.

> or provide for a different mechanism for assertions after exit() is prevented?

If you are still interested, could you expand on this a bit? I'm not sure I understand your use case.

**Comment by bysse** [comment]
Thanks for an excellent sysexit workaround! I have the same use case. 
I'm doing some "end-to-end" testing of a CLI tool that produces output to the filesystem. 
So basically I'm catching the SecurityException in the test method and then validate the results on the filesystem.
It feels more correct if SystemExitPreventedException was public so you don't have to catch the more generic SecurityException. A bonus would be to add the exit code to that exception.

**Comment by tginsberg** [comment]
This has been done in release 1.1.0, which should now be in Maven Central. Thanks for the suggestion!

**Comment by tginsberg** [comment]
This was fixed in 1.1.0, but use 1.1.1 instead. There was an unrelated but in 1.1.0.

**Comment by bysse** [comment]
Thanks a lot!
